File: mySolutions/128. Longest Consecutive Sequence/128. Longest Consecutive Sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }
        
        Map<Integer, Integer> map_num_length = new HashMap<>();
        int maxLength = 0;
        for (int n : nums) {
            
            if (map_num_length.containsKey(n)) {
                continue;
            } 
            else {
                int leftLength = map_num_length.containsKey(n - 1) ? map_num_length.get(n - 1) : 0;
                int rightLength = map_num_length.containsKey(n + 1) ? map_num_length.get(n + 1) : 0;
                
                int sum = 1 + leftLength + rightLength;
                
                // System.out.printf("n: %d, leftLength: %d, rightLength: %d, sum: %d\n", n, leftLength, rightLength, sum);
                
                maxLength = Math.max(sum, maxLength);
                
                map_num_length.put(n, sum);
                map_num_length.put(n - leftLength, sum);
                map_num_length.put(n + rightLength, sum);
                
            }
            
        }
        
        return maxLength;
        
    }
}

File: mySolutions/128. Longest Consecutive Sequence/128. Longest Consecutive Sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        map = {}
        maxLen = 0
        
        for num in nums:
          if num in map:
            continue
          
          leftMost = map.get(num - 1, 0)
          rightMost = map.get(num + 1, 0)
          
          sum = 1 + leftMost + rightMost
          
          if num - leftMost in map: map[num - leftMost] = sum
          if num + rightMost in map: map[num + rightMost] = sum
          
          map[num] = sum
          
        for key in map:
          maxLen = max(map[key], maxLen)
          
        return maxLen